### PR TITLE
New gatling

### PR DESCRIPTION
**Pull request by 19thanks96**
[change gatling: add bullets with vx, vy and collision](https://github.com/DarthVanger/ghost-of-kyiv/issues/94)

### CODE REVIEW

File: enemy.js [hunk 1]
```diff
@@ -46,6 +46,7 @@ function createEnemy(src, width, height, i, maxHealth) {
   let enemy = {};
   enemy.element = document.createElement('img')
   enemy.element.id = 'enemy' + i;
+  enemy.element.classNmae = 'enemy'
```
Comment (DarthVanger) [inline enemy.js:49]: `classNmae` is misspelled.
It means `className` is never set, and is never used.
Delete this line.

File: gatling.js [hunk 2]
```diff
@@ -12,26 +13,42 @@ export let gatling = {
   element: document.querySelector("#gatling"),
 };
 
-export function fireGatlingEnemy (event, enemy) {  
-    if (airfighter.x + airfighter.width < event.pageX ) { 
-      enemy.enemyHealth.element.value -= gatling.dmg;
-      gatling.ammo -= 10;
+export let bulletArray = []
+
+
```
Comment (DarthVanger) [inline gatling.js:18]: Delete 3 enters :) 
Leave only 1

File: gatling.js [hunk 3]
```diff
@@ -12,26 +13,42 @@ export let gatling = {
...
+
+function createBullet() {
+  let bullet = {}
```
Comment (DarthVanger) [inline gatling.js:21]: Use `const`

File: gatling.js [hunk 4]
```diff
@@ -12,26 +13,42 @@ export let gatling = {
...
+  bullet.x = airfighter.x + airfighter.width + (Math.random()* 4)
+  bullet.y = airfighter.y + airfighter.height/2
+  bullet.margin = Math.random()*(3)-(1.5)
+  gatling.ammo -= 10;
+  bulletArray.push(bullet)
+}
+
+export function removeBullet(bullet) {
+  bullet.element.remove()
+  bulletArray = bulletArray.filter(anothetBullet => anothetBullet !== bullet)
```
Comment (DarthVanger) [inline gatling.js:37]: Misspelled word: `anothetBullet` -> `anotherBullet`

File: gatling.js [hunk 5]
```diff
@@ -12,26 +13,42 @@ export let gatling = {
...
+}
+
+
```
Comment (Alianshark) [inline gatling.js:40]: delete one enter of two

File: gatling.js [hunk 6]
```diff
@@ -12,26 +13,42 @@ export let gatling = {
...
+export function fireGatlingEnemy (event, enemy) {
+  if(gatling.ammo > 0) {
```
Comment (Alianshark) [inline gatling.js:42]: needs space after if

File: gatling.js [hunk 7]
```diff
@@ -12,26 +13,42 @@ export let gatling = {
...
+    if(bulletArray.length < 10) {
```
Comment (Alianshark) [inline gatling.js:43]: the same, need space after if

File: step.js [hunk 8]
```diff
@@ -124,6 +125,40 @@ function moveEnemy(enemy) {
 }
 
 
+
+function moveBullet(bullet) {
+  bullet.x += bullet.velocity
+  bullet.y += bullet.margin
+    bullet.element.style.left = bullet.x + 'px'
```
Comment (Alianshark) [inline step.js:132]: not needed space

File: step.js [hunk 9]
```diff
@@ -124,6 +125,40 @@ function moveEnemy(enemy) {
...
+  bulletArray.forEach((bullet) => checkBulletCollision(bullet, enemy))
+}
+
+function checkBulletCollision(bullet, enemy) {
+    const isCollision = checkCollision(bullet, enemy)
+    const isOutOfScreen = bullet.x > window.innerWidth
+    if (isCollision) {
+      enemy.enemyHealth.element.value -= 5
+    }
+    if(isOutOfScreen || isCollision) {
```
Comment (Alianshark) [inline step.js:144]: need space after if

File: step.js [hunk 10]
```diff
@@ -124,6 +125,40 @@ function moveEnemy(enemy) {
...
+
+function moveBullet(bullet) {
+  bullet.x += bullet.velocity
+  bullet.y += bullet.margin
+    bullet.element.style.left = bullet.x + 'px'
+    bullet.element.style.top = bullet.y + 'px'
+}
+
+function enemyCollisionWithBullet(enemy) {
+  bulletArray.forEach((bullet) => checkBulletCollision(bullet, enemy))
```
Comment (Alianshark) [inline step.js]: you can leave only one bracket after forEach befro bullet

File: step.js [hunk 11]
```diff
@@ -124,6 +124,37 @@ function moveEnemy(enemy) {
...
+
+function checkBulletCollision(bullet, enemy) {
+    const isCollision = checkCollision(bullet, enemy)
+    const isOutOfScreen = bullet.x > window.innerWidth
+    if (isCollision) {
+      enemy.enemyHealth.element.value -= 5
+    }
+    if (isOutOfScreen || isCollision) {
+      removeBullet(bullet)
+  }
```
Comment (Alianshark) [inline step.js:146]: something strange with brackets

**Comment by DarthVanger** [comment]
OGOGN 🔥🔥🔥

**Review by DarthVanger — APPROVED** [review]

**Review by Alianshark — APPROVED** [review]
LGTM

**Pull request merged**